ako-gateway-api/k8s/validator.go

```diff
@@ -378,27 +366,67 @@ func validateParentReference(key string, httpRoute *gatewayv1.HTTPRoute, httpRou
 		listenersForRoute = append(listenersForRoute, gateway.Spec.Listeners...)
 	}
 
+	// TODO: Validation for hostname (those being fqdns) need to validate as per the K8 gateway req.
 	var listenersMatchedToRoute []gatewayv1.Listener
 	for _, listenerObj := range listenersForRoute {
 		// check from store
 		hostInListener := listenerObj.Hostname
+		isListenerFqdnWildcard := false
+		matched := false
+		// TODO:
+		// Use case to handle for validations of hostname:
+		// USe case 1: Shouldn't contain mor than 1 *
+		// USe case 2: * should be at the beginning only
+		if hostInListener == nil || *hostInListener == "" || *hostInListener == utils.WILDCARD {
+			matched = true
+		} else {
+			// mark listener fqdn if it has *
+			if strings.HasPrefix(string(*hostInListener), utils.WILDCARD) {
+				isListenerFqdnWildcard = true
+			}
+			for _, host := range httpRoute.Spec.Hostnames {
+				// casese to consider:
+				// Case 1: hostname of gateway is wildcard(empty) and hostname from httproute is not wild card
+				// Case 2: hostname of gateway is not wild card and hostname from httproute is wildcard
+				// case 3: hostname of gateway is wildcard(empty) and hostname from httproute is wildcard
+				// case 4: hostname of gateway is not wildcard and hostname from httproute is not wildcard
+				isHttpRouteHostFqdnWildcard := false
+				if strings.HasPrefix(string(host), utils.WILDCARD) {
+					isHttpRouteHostFqdnWildcard = true
+				}
+				if isHttpRouteHostFqdnWildcard && isListenerFqdnWildcard {
+					// both are true. Match nonwildcard part
+					nonWildCardHttpRouteHostname := strings.Split(string(host), utils.WILDCARD)
+					nonWildCardGWHostname := strings.Split(string(*hostInListener), utils.WILDCARD)
+					// Use case: 1. GW: *.avi.internal HttpRoute: *.bar.avi.internal
+					// USe case: 2. GW: *.bar.avi.internal HttpRoute: *.avi.internal
```

Comment:
Is this splitting and sending only nonWildCard part of hostnames to CheckSubdomainOverlapping even required ?
As far as I understand, CheckSubdomainOverlapping itself can handle that

Reply:
done,